internal/pkg/containers/image/resolver.go

```diff
@@ -97,25 +98,54 @@ func RegistryHosts(reg config.Registries) docker.RegistryHosts {
 	}
 }
 
+// EndpointEntry represents a registry endpoint.
+type EndpointEntry struct {
+	Endpoint     string
+	OverridePath bool
+}
+
+// RegistryEndpointEntriesFromConfig returns registry endpoints per host.
+func RegistryEndpointEntriesFromConfig(host string, reg config.RegistryMirrorConfig) ([]EndpointEntry, error) {
+	entries := xslices.Map(reg.Endpoints(), func(endpoint string) EndpointEntry {
+		return EndpointEntry{Endpoint: endpoint, OverridePath: reg.OverridePath()}
+	})
+
+	if !reg.SkipFallback() {
```

Comment:
nit: just return when skipping to reduce code nesting?

Reply:
yeah good idea, thanks, I will update!